File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        r, c = len(matrix), len(matrix[0])
        rows, cols = [False]*r, [False]*c

        # mark all rows & cols that need to be changed
        for i in range(r):
            for j in range(c):
                if matrix[i][j] == 0:
                    rows[i] = cols[j] = True

        # change all cols for a row
        for i in range(r):
            if rows[i]:
                for j in range(c): matrix[i][j] = 0

        # change all rows for a col
        for j in range(c):
            if cols[j]:
                for i in range(r): matrix[i][j] = 0
        

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cs
public class Solution {
    public void SetZeroes(int[][] matrix) {
        int r = matrix.Length, c = matrix[0].Length;
        bool[] rows = new bool[r], cols = new bool[c];

        // mark all rows & cols that need to be changed
        for(var i=0; i<r; i++)
            for(var j=0; j<c; j++)
                if(matrix[i][j] == 0)
                    rows[i] = cols[j] = true;

        // change all cols for a row
        for(var i=0; i<r; i++)
            if(rows[i])
                for(var j=0; j<c; j++) matrix[i][j] = 0;

        // change all rows for a col
        for(var i=0; i<c; i++)
            if(cols[i])
                for(var j=0; j<r; j++) matrix[j][i] = 0;
    }
}